src/parsers/perf/perfparser.cpp

```diff
@@ -1418,71 +1487,183 @@ void PerfParser::startParseFile(const QString& path, const QString& sysroot, con
         }
 
         QProcess process;
-        auto env = Util::appImageEnvironment();
 
-        const auto customUrls = Settings::instance()->debuginfodUrls();
-        if (!customUrls.isEmpty()) {
-            const auto envVar = QStringLiteral("DEBUGINFOD_URLS");
-            const auto defaultUrls = env.value(envVar);
-            const auto separator = QLatin1Char(' ');
-            env.insert(envVar, customUrls.join(separator) + separator + defaultUrls);
+        process.setProcessEnvironment(env);
+        process.setProcessChannelMode(QProcess::ForwardedErrorChannel);
+        connect(this, &PerfParser::stopRequested, &process, &QProcess::kill);
+
+        d.setInput(&process);
+
+        const auto exitCodeHandler = [finalize, this](int exitCode, QProcess::ExitStatus exitStatus) {
```

Comment:
this looks like duplicated code? why? ideally please restructure this such that you first refactor the code to enable diffing, but without actually implementing that part. that will make this series much easier for me to review. and we could push the cleanup/refactoring patches as soon as they are ready without having to wait for the whole series to be finalized

Reply:
having looks more at this code, I think there is a much simpler and cleaner approach to get the diffing implemented:

Basically the idea is as follows:

```
void PerfParser::startParseFile(const QString& path, const QString& diffFile)
{
    // obviously only when diffFile is not empty
    PerfParser otherData;
    otherData.startParseFile(diffFile);

    // do the normal dance here for the path

    // then wait for both parsers (this and otherData) to be ready
    // then finally subtract the data
}
```
```